ogr/ogrsf_frmts/miramon/ogrmiramondatasource.cpp

```diff
@@ -0,0 +1,310 @@
+    if (!poLayer->bValidFile)
+    {
+        delete poLayer;
+        return FALSE;
+    }
+    papoLayers = static_cast<OGRMiraMonLayer **>(CPLRealloc(
+        papoLayers,
+        (size_t)(sizeof(OGRMiraMonLayer *) * ((size_t)nLayers + (size_t)1))));
+    papoLayers[nLayers] = poLayer;
+    nLayers++;
+
+    if (pszDSName)
+    {
+        const char *pszExtension = CPLGetExtension(pszDSName);
+        if (!EQUAL(pszExtension, "pol") && !EQUAL(pszExtension, "arc") &&
+            !EQUAL(pszExtension, "pnt"))
+        {
+            strcpy(
+                MMMap.pszMapName,
+                CPLFormFilename(pszDSName, CPLGetBasename(pszDSName), "mmm"));
+            if (!MMMap.nNumberOfLayers)
+            {
+                MMMap.fMMMap = VSIFOpenL(MMMap.pszMapName, "w+");
+                VSIFPrintfL(MMMap.fMMMap, "[VERSIO]\n");
+                VSIFPrintfL(MMMap.fMMMap, "Vers=2\n");
+                VSIFPrintfL(MMMap.fMMMap, "SubVers=0\n");
+                VSIFPrintfL(MMMap.fMMMap, "variant=b\n");
+                VSIFPrintfL(MMMap.fMMMap, "\n");
+                VSIFPrintfL(MMMap.fMMMap, "[DOCUMENT]\n");
+                VSIFPrintfL(MMMap.fMMMap, "Titol= %s(map)\n",
+                            CPLGetBasename(poLayer->GetName()));
+                VSIFPrintfL(MMMap.fMMMap, "\n");
+            }
+        }
+        else
+            *MMMap.pszMapName = '\0';
+    }
+    else
+        *MMMap.pszMapName = '\0';
+
+    if (pszDSName)
+        CPLFree(pszDSName);
+    pszDSName = CPLStrdup(pszFilename);
+
+    return TRUE;
+}
+
+/****************************************************************************/
+/*                               Create()                                   */
+/*                                                                          */
+/*      Create a new datasource.  This does not really do anything          */
+/*      currently but save the name.                                        */
+/****************************************************************************/
+
+int OGRMiraMonDataSource::Create(const char *pszDataSetName,
+                                 char ** /* papszOptions */)
+
+{
+    pszDSName = CPLStrdup(pszDataSetName);
+    pszRootName = CPLStrdup(pszDataSetName);
+
+    return TRUE;
+}
+
+/****************************************************************************/
+/*                           ICreateLayer()                                 */
+/****************************************************************************/
+
+OGRLayer *
+OGRMiraMonDataSource::ICreateLayer(const char *pszLayerName,
+                                   const OGRGeomFieldDefn *poGeomFieldDefn,
+                                   CSLConstList papszOptions)
+{
+    CPLAssert(nullptr != pszLayerName);
+
+    const auto eType = poGeomFieldDefn ? poGeomFieldDefn->GetType() : wkbNone;
+    const auto poSRS =
+        poGeomFieldDefn ? poGeomFieldDefn->GetSpatialRef() : nullptr;
+
+    switch (eType)
```

Comment:
This whole switch() can be just replaced by a ``if (OGR_GT_HasM(eType))`` check

Reply:
MUCH better ;)